### PR TITLE
Vue 3.0 upgrade

### PR DESCRIPTION
**Pull request by softchris**
- upgraded code
- upgraded various config files

issues: 
- it compiles, currently there's a 502 error, gateway issue, probably proxy
- resolve is ../ instead of @/ due to conflict with @fortawesome when trying to install  (npm install --save-dev eslint-import-resolver-alias)

### CODE REVIEW

**Review by johnpapa — COMMENTED** [review]
Why did you change the @click to handlers? @softchris

**Review by johnpapa — COMMENTED** [review]
> Why did you change the @click to handlers? @softchris 

>   Why did you add this for https? 

```
devServer: {
    server: {
      type: "https"
    },
```

**Review by johnpapa — COMMENTED** [review]
A few changes:

1. I udpated 2 more npm packages in the package.json file as it could not npminstall without them. Plase review.
2. I removed the https as it caused issues for me locally. This helped me with running the app with Functions and SWA. (no more proxy issues)
3. I updated the npm scripts to start the app using the correct embedded quotes with the SWA CLI.
4. I added an .env file ... not sure if it is needed.

**Comment by softchris** [comment]
> devServer: {
>     server: {
>       type: "https"
>     },

@johnpapa cause otherwise it would show up like this in Codespaces, 

```
Failed to construct 'WebSocket': An insecure WebSocket connection may not be initiated from a page loaded over HTTPS.
    at new WebSocketClient (https://orange-system-5rp49wqrp934jgq-8080.app.github.dev/js/chunk-vendors.js:17266:19)
    at initSocket (https://orange-system-5rp49wqrp934jgq-8080.app.github.dev/js/chunk-vendors.js:19061:12)
    at ./node_modules/webpack-dev-server/client/index.js?protocol=ws&hostname=172.16.5.4&port=8080&pathname=%2Fws&logging=none&progress=true&overlay=%7B%22errors%22%3Atrue%2C%22warnings%22%3Afalse%7D&reconnect=10&hot=true&live-reload=true (https://orange-system-5rp49wqrp934jgq-8080.app.github.dev/js/chunk-vendors.js:17646:55)
    at __webpack_require__ (https://orange-system-5rp49wqrp934jgq-8080.app.github.dev/js/app.js:1557:32)
    at https://orange-system-5rp49wqrp934jgq-8080.app.github.dev/js/app.js:2730:83
    at __webpack_require__.O (https://orange-system-5rp49wqrp934jgq-8080.app.github.dev/js/app.js:1599:23)
    at https://orange-system-5rp49wqrp934jgq-8080.app.github.dev/js/app.js:2733:53
    at https://orange-system-5rp49wqrp934jgq-8080.app.github.dev/js/app.js:2735:12
```

**Comment by softchris** [comment]
> Why did you change the @click to handlers? @softchris

@johnpapa  if you are talking about the added keyhandlers then it was a linting rule that said for accessibility reasons you should have a keyhandler where you add a click handler on non button elements. I can shut off that linting rule if you want? @johnpapa

File: vue-app/.eslintrc.js [hunk 1]
```diff
@@ -26,8 +26,11 @@ module.exports = {
         ignoreWhenBindingPresent: true,
       },
     ],
+    'vuejs-accessibility/label-has-for': 'off',
+    'vue/multi-word-component-names':'off',
+    'no-restricted-exports': 'off',
   },
   parserOptions: {
-    parser: 'babel-eslint',
+    parser: '@babel/eslint-parser',
```
Comment (softchris) [inline vue-app/.eslintrc.js:34]: FYI: eslint-parser replaces babel-eslint

File: vue-app/babel.config.js [hunk 2]
```diff
@@ -1,3 +1,3 @@
-module.exports = {
-  presets: ['@vue/cli-plugin-babel/preset'],
-};
+// module.exports = {
```
Comment (softchris) [inline vue-app/babel.config.js:1]: if app seems to work, then I recommend we remove this file

File: vue-app/src/components/auth-login.vue [hunk 3]
```diff
@@ -14,10 +14,15 @@ export default {
       const url = `/.auth/login/${this.provider}?${redirect}`;
       window.location.href = url;
     },
+    handleKeyDown() {
```
Comment (softchris) [inline vue-app/src/components/auth-login.vue:17]: added due to lint rule stating that non button element with click handlers need to have keyhandler for accessibility, I can disable this rule or keep this code, please let me know which @johnpapa

File: vue-app/src/components/auth-login.vue [hunk 4]
```diff
@@ -14,10 +14,15 @@ export default {
...
+      this.goAuth();
+    },
   },
 };
 </script>
 
 <template>
-  <div class="auth-link" @click="goAuth">{{ provider }}</div>
+  <div class="auth-link" @click="goAuth" @keydown.enter="handleKeyDown">
```
Comment (softchris) [inline vue-app/src/components/auth-login.vue:25]: added due to lint rule stating that non button element with click handlers need to have keyhandler for accessibility, I can disable this rule or keep this code, please let me know which @johnpapa

File: vue-app/src/components/auth-logout.vue [hunk 5]
```diff
@@ -8,10 +8,15 @@ export default {
       const url = `/.auth/logout?${redirect}`;
       window.location.href = url;
     },
+    handleKeyDown() {
```
Comment (softchris) [inline vue-app/src/components/auth-logout.vue:11]: added due to lint rule stating that non button element with click handlers need to have keyhandler for accessibility, I can disable this rule or keep this code, please let me know which @johnpapa

File: vue-app/src/components/auth-logout.vue [hunk 6]
```diff
@@ -8,10 +8,15 @@ export default {
...
+      this.goAuth();
+    },
   },
 };
 </script>
 
 <template>
-  <div class="auth-link" @click="goAuth">Logout</div>
+  <div class="auth-link" @click="goAuth" @keydown.enter="handleKeyDown">
```
Comment (softchris) [inline vue-app/src/components/auth-logout.vue:19]: added due to lint rule stating that non button element with click handlers need to have keyhandler for accessibility, I can disable this rule or keep this code, please let me know which @johnpapa

File: vue-app/src/components/button-footer.vue [hunk 7]
```diff
@@ -43,6 +44,7 @@ export default {
     @click="handleClick"
     :data-index="dataIndex"
     :data-id="dataId"
+    type="button"
```
Comment (softchris) [inline vue-app/src/components/button-footer.vue:47]: added due to lint rule on accessibility, i.e must have a type, keep or remove? @johnpapa

File: vue-app/src/components/list-header.vue [hunk 8]
```diff
@@ -32,6 +32,7 @@ export default {
       <h2 class="title">{{ title }}</h2>
     </router-link>
     <button
+      type="button"
```
Comment (softchris) [inline vue-app/src/components/list-header.vue:35]: added due to lint rule on accessibility, i.e must have a type, keep or remove? @johnpapa

File: vue-app/src/components/nav-bar.vue [hunk 9]
```diff
@@ -49,9 +49,11 @@ export default {
       <p class="menu-label">Auth</p>
       <div class="menu-list auth">
         <template v-if="!userInfo">
-          <template v-for="provider in providers">
-            <AuthLogin :key="provider" :provider="provider" />
-          </template>
+          <AuthLogin
```
Comment (softchris) [inline vue-app/src/components/nav-bar.vue:52]: removed due to lint rule in Vue, i.e said it doesn't need to be wrapped in a template like that

File: vue-app/src/router.js [hunk 10]
```diff
@@ -1,43 +1,40 @@
-import Vue from 'vue';
-import Router from 'vue-router';
-import PageNotFound from '@/components/page-not-found.vue';
+import { createRouter, createWebHistory } from 'vue-router';
+import PageNotFound from './components/page-not-found.vue';
 
-Vue.use(Router);
+const routes = [
```
Comment (softchris) [inline vue-app/src/router.js:4]: Vue 3 way of bootstrapping router

File: vue-app/src/views/home.vue [hunk 11]
```diff
@@ -9,19 +9,19 @@
       <p>Log in to start enjoying your benefits</p>
       <br />
       <div class="button-group">
-        <button class="button" aria-label="My List" tabindex="0">
+        <button class="button" type="button" aria-label="My List" tabindex="0">
```
Comment (softchris) [inline vue-app/src/views/home.vue:12]: lint rule on accessibility

File: vue-app/.eslintrc.js (hunk 1 above)
Comment (johnpapa) [reply vue-app/.eslintrc.js:34]: ok

File: vue-app/babel.config.js (hunk 2 above)
Comment (johnpapa) [reply vue-app/babel.config.js:1]: I removed the file. but do we need to remove anything from the `package.json` too?

**Review by johnpapa — APPROVED** [review]

**Pull request merged**